### ISSUE TITLE
[Bug] 2-setup-azure-environment.yml Error

### ISSUE THREAD

**Issue by shearpaladin** [skills/deploy-to-azure#47]
### Summary

The environment should use 'with' not 'env' from line 84 to line 100. I kept facing this issue wondering why environment variables weren't being set

### How to reproduce
Prereq: On Mac

1) Following all Instructions beginning this Lab
2) Upon Editing the deployment file with your username Github Actions will run and give you this Error

### Screenshots
<img width="695" alt="Screenshot 2023-06-04 at 1 05 17 PM" src="https://github.com/skills/deploy-to-azure/assets/46476399/2c2fcded-baef-4241-b154-757fc8237197">

### SOLUTION
I changed the lines 84 - 100 for 'env' -> 'with' in 2-setup-azure-environment.yml and it resolved the issue.

<img width="1007" alt="Screenshot 2023-06-04 at 1 32 44 PM" src="https://github.com/skills/deploy-to-azure/assets/46476399/4678256a-16a0-49bb-964e-83bea973c87f">

**Comment by shearpaladin** [comment]
This was also resolved with #38 by changing the path in pull-main.sh to initialize-repository.sh

**Comment by yl2049** [comment]
> This was also resolved with #38 by changing the path in pull-main.sh to initialize-repository.sh

I had a permission denied error after changing the name to ```initialize-repository.sh```. 
I resolved it by adding an extra command 
```
      - name: Pull main
        run: |
          chmod +x ./.github/script/initialize-repository.sh
          ./.github/script/initialize-repository.sh
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by hectorsector** [comment]
> This was also resolved with [#38](https://github.com/skills/deploy-to-azure/issues/38) by changing the path in pull-main.sh to initialize-repository.sh

Sounds like this solved it so I'll close it out but please reopen if I misunderstood.
